src/ffi/array.rs

```diff
@@ -235,9 +235,14 @@ unsafe fn create_buffer<T: NativeType>(
     owner: InternalArrowArray,
     index: usize,
 ) -> Result<Buffer<T>> {
+    let len = buffer_len(array, data_type, index)?;
+
+    if len == 0 {
```

Comment:
c++ arrow can create null pointers when arrays are empty. This ensures we don't have to deal with that.